arch/arm/core/aarch32/swap_helper.S

```diff
@@ -498,8 +498,29 @@ _stack_frame_endif:
 
 _oops:
     push {r0, lr}
+#if defined(CONFIG_EXTRA_EXCEPTION_INFO)
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+    /* Build _callee_saved_t. To match the struct
+     * definition we push the psp & then r11-r4
+     */
+    mrs r1, PSP
+    push {r1}
+    push {r4-r11}
+    mov  r1, sp /* pointer to _callee_saved_t */
+#endif /* CONFIG_ARMV7_M_ARMV8_M_MAINLINE */
+#endif /* CONFIG_EXTRA_EXCEPTION_INFO */
     bl z_do_kernel_oops
```

Comment:
This seems to break the 8-byte stack pointer alignment across function calls required by the AAPCS (i.e. when `z_do_kernel_oops` is called, the stack pointer will not be 8-byte aligned).

Should align the stack pointer before calling `z_do_kernel_oops`.